### DISCUSSION TITLE
greengras-bin to RDEPENDS on ntp-systemd

### DISCUSSION THREAD

**Discussion by pwillem** [aws4embeddedlinux/meta-aws#3242] [Q&A] [answered]
Hello

I'm integrating greengrass-bin in an image that includes systemd and chrony for time synchronization. The chrony recipe is incompatible with [NTP](https://github.com/openembedded/meta-openembedded/blob/master/meta-networking/recipes-support/chrony/chrony_4.3.bb#L131-L136) which [greengrass-bin_2.9.5.bb](https://github.com/aws4embeddedlinux/meta-aws/blob/master-next/recipes-iot/aws-iot-greengrass/greengrass-bin_2.9.5.bb#L45-L48) has as RDEPENDS.

Is there a reason for greengrass to RDEPENDS on ntp when using systemd? I couldn't find a reference to ntp in [greengrass nucleus](https://github.com/aws-greengrass/aws-greengrass-nucleus) and it seems chrony provides the same functionality.

**Comment by thomas-roos** [comment, marked as answer]
Hi,
you are right - there is no reason to hard depend on ntp when using systemd.
The only reason is that you need a correct clock to connect to aws-iot-core (cert check etc.).
Will change this RDEPENDS to RRECOMMENDS and add a comment.
Also moving this to greengrass-bin-demo.
